# HG changeset patch
# User Rik <rik@octave.org>
# Date 1389505377 28800
#      Sat Jan 11 21:42:57 2014 -0800
# Node ID c6858f725afe36d7e70fe1d76b943b28f7bf8d43
# Parent  efaff9f3ca396ae91ebcd5f14493e71275da0010
Use std::string rather than for loop to generate blank string for indentation.

* pt-pr-code.cc (tree_print_code::indent): Use std::string constructor to
generate a blank string rather than for loop for ~7% performance improvement.

diff --git a/libinterp/parse-tree/pt-pr-code.cc b/libinterp/parse-tree/pt-pr-code.cc
--- a/libinterp/parse-tree/pt-pr-code.cc
+++ b/libinterp/parse-tree/pt-pr-code.cc
@@ -1195,32 +1195,28 @@ tree_print_code::print_fcn_handle_body (
               suppress_newlines++;
               c->accept (*this);
               suppress_newlines--;
             }
         }
     }
 }
 
-// Each print_code() function should call this before printing
-// anything.
-//
-// This doesn't need to be fast, but isn't there a better way?
+// Each print_code() function should call this before printing anything.
 
 void
 tree_print_code::indent (void)
 {
   assert (curr_print_indent_level >= 0);
 
   if (beginning_of_line)
     {
       os << prefix;
 
-      for (int i = 0; i < curr_print_indent_level; i++)
-        os << " ";
+      os << std::string (curr_print_indent_level, ' ');
 
       beginning_of_line = false;
     }
 }
 
 // All print_code() functions should use this to print new lines.
 
 void
